torchtune/datasets/_text_completion.py

```diff
@@ -44,6 +44,9 @@ def __init__(
         column: str = "text",
         max_seq_len: Optional[int] = None,
         add_eos: bool = True,
+        custom_filter: bool=False,
+        skip_filter: bool = False,
+        filter_fn: Optional[Callable] = None,
```

Comment:
while I see that this may provide more flexibility, as a new user it might be overwhelming to see three filtering parameters and understand how they work together. I would advise going with the simplest approach first of just passing in an optional filter_fn and only apply it if it is specified.

On if there should be a default filter, that's an interesting point. Personally, I would lean towards having the default be none, so we're not removing data samples without the user knowing and forces them to have the explicit intention of modifying their data. In the dataset builders themselves for specific datasets it's more reasonable to set `filter_fn` to something like what you defined because we already know that this dataset needs some preprocessing.